src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -957,7 +960,7 @@ protected void ExecuteIndexing()
 
                                 if (numberOfSetEvents == 1 && _logsAppliedEvent.IsSet)
                                 {
-                                    _hadRealIndexingWorkToDo = false;
+                                    _hadRealIndexingWorkToDo.LowerOrDie();
```

Comment:
I'm concerned with the `XxxOrDie` here. The previous code didn't have issues with calling this multiple times.
Haven't actually got to these methods yet, but I'm concerned that this can be called twice and kill us.
In general, there is nothing that should prevent us from setting `_hadRealIndexingWorkToDo` to `false` twice.

Reply:
Replaced all .RaiseOrDie or .LowerOrDie for their not dying alternatives (which was the original behavior in all places anyways). Functions are left there so that people may start implementing with stronger invariants in mind.